.github/workflows/publish-stack-set-to-service-catalog.yaml

```diff
@@ -58,6 +58,23 @@ jobs:
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
+      - name: Assume AWS Role
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ secrets.DT_GH_ACTIONS_ROLE_ARN }}
+          aws-region: eu-west-2
+          role-duration-seconds: 7200
+      - name: Search latest hardened AMI
+        id: hardened-ami
+        run: |
+          LATEST-AMI=$(aws ec2 describe-images \
+          --owners aws-marketplace \
+          --filters 'Name=name,Values=*'${PRODUCT_ID}'*' \
+          --query 'Images[*].[ImageId,CreationDate]' --output text | sort -k2 -r | head -n1 | awk '{ print $1 }')
+          echo "amiId=$LATEST_AMI" >> $GITHUB_OUTPUT
+        env:
+          AWS_DEFAULT_REGION: eu-west-2
+          PRODUCT_ID: "c41d38c4-3f6a-4434-9a86-06dd331d3f9c"
       - name: cdk synth
         uses: youyo/aws-cdk-github-actions@v2
```

Comment:
Where does this come from?
Discuss if we should use this action.
Pin the action version, at the very least.